packages/moonstone/Scroller/Scroller.js

```diff
@@ -298,49 +298,55 @@ class ScrollerBase extends Component {
 		}
 	}
 
-	getNextEndPoint = (direction, oSpotBounds) => {
-		const bounds = this.uiRefCurrent.getScrollBounds();
-
+	getNextEndPoint = (direction, scrollerBounds, focusedItemBounds, pageDiff) => {
 		let oPoint = {};
 		switch (direction) {
 			case 'up':
-				oPoint.x = oSpotBounds.left;
-				oPoint.y = oSpotBounds.top - bounds.clientHeight;
+				oPoint.x = focusedItemBounds.left;
+				oPoint.y = scrollerBounds.top - scrollerBounds.height * (pageDiff);
 				break;
 			case 'left':
-				oPoint.x = oSpotBounds.left - bounds.clientWidth;
-				oPoint.y = oSpotBounds.top;
+				oPoint.x = focusedItemBounds.left - scrollerBounds.width;
+				oPoint.y = focusedItemBounds.top;
 				break;
 			case 'down':
-				oPoint.x = oSpotBounds.left;
-				oPoint.y = oSpotBounds.top + oSpotBounds.height + bounds.clientHeight;
+				oPoint.x = focusedItemBounds.left;
+				oPoint.y = scrollerBounds.top + scrollerBounds.height * (pageDiff + 1);
 				break;
 			case 'right':
-				oPoint.x = oSpotBounds.left + oSpotBounds.width + bounds.clientWidth;
-				oPoint.y = oSpotBounds.top;
+				oPoint.x = focusedItemBounds.left + focusedItemBounds.width + scrollerBounds.width;
```

Comment:
I'm really confused about the rule for this case here.

Reply:
We don't need the `right` and `left` cases. So I removed those cases.